pymc_marketing/mmm/lift_test.py

```diff
@@ -0,0 +1,366 @@
+"""Lift test functions for the MMM."""
+
+from functools import partial
+from typing import Callable, Optional
+
+import numpy as np
+import pandas as pd
+import pymc as pm
+import pytensor.tensor as pt
+from numpy import typing as npt
+
+from pymc_marketing.mmm.transformers import logistic_saturation
+from pymc_marketing.mmm.utils import michaelis_menten
+
+
+class MissingLiftTestError(Exception):
+    def __init__(self, missing_values: npt.NDArray[np.int_]) -> None:
+        self.missing_values = missing_values
+        super().__init__(
+            f"Some lift test values are not in the model: {missing_values}"
+        )
+
+
+def _lift_test_index(
+    lift_values: np.ndarray, model_values: np.ndarray
+) -> npt.NDArray[np.int_]:
```

Comment:
I think we should keep the same type hit convention for np arrays `np.ndarray` or `npt.NDArray`. Mixing them look confusing ;)